core/src/test/scala/org/apache/spark/util/UtilsSuite.scala

```diff
@@ -924,6 +926,24 @@ class UtilsSuite extends SparkFunSuite with ResetSystemProperties with Logging {
     }
   }
 
+  test("test class ClassCastException") {
```

Comment:
This test does not seem to be testing Spark code? I don't understand.

Reply:
> This test does not seem to be testing Spark code? I don't understand.

If no exception is thrown, it will run into line 938 `assert(false)` and this test will not pass. But it always go into `catch`, throw `ClassCastException `, so this test pass. Actually, i

Reply:
Yes, but you're not testing anything in Spark, so I don't think this is useful. You're just copying the same code and asserting that it actually still fails before the change.

Existing tests don't catch this case?

Reply:
> Yes, but you're not testing anything in Spark, so I don't think this is useful. You're just copying the same code and asserting that it actually still fails before the change.
> 
> Existing tests don't catch this case?

I try to read some use of `Utils.classForName`,  and I found there are about two usage scenarios: 
1. load class and not need to `getConstructor().newInstance()`
2. need to `getConstructor().newInstance()`
In second case, usually use ` :Class[_]`.

Reply:
> Yes, but you're not testing anything in Spark, so I don't think this is useful. You're just copying the same code and asserting that it actually still fails before the change.
> 
> Existing tests don't catch this case?

Yes, please review the new change, and I will delete this test. I just want to show you that it's always thrown exception.